Assignment/Test.c

```diff
@@ -4,11 +4,143 @@
+      printf (" |%*s%20s%*s   |\n", padding, "", result, padding, "");
+      free (actualOutput);
+   }
+   printf ("\n\t\t\t\tTest Cases for Binary Search\n\n"
+           "\tInput\t\t\t\tOutput\t\t  Key\t\tResult\n |--------------------------|"
+           "--------------------------|------|------------------------|\n");
+   for (int i = 0; i < numOfTestCases; i++) {
+      int* actualOutput = malloc (size[i] * sizeof (int));
+      PrintTable (input[i], size[i], 6);
+      PrintTable (expOutput[i], size[i], 6);
+      int Keys[] = { 3, 25, -1, 65, 3, -10 },
+         foundIndex = BinarySearch (expOutput[i], size[i], Keys[i]);
+      printf (" | %-3d  | ", Keys[i]);
+      printf (foundIndex != -1 ? GREEN_TEXT "Element %3d found at %d" RESET_TEXT
+              : RED_TEXT "Element %3d Not Found " RESET_TEXT, Keys[i], foundIndex);
+      printf (" |\n");
+      free (actualOutput);
+   }
+}
+
+void UserInput () {
+   int arr[100] = { 0 }, n;
+   char buffer[256], * endptr;
+   while (1) {
+      printf ("Enter number of elements: ");
+      fgets (buffer, sizeof (buffer), stdin);
+      n = strtol (buffer, &endptr, 10);
+      // Check for valid input
+      if (endptr != buffer && *endptr == '\n' && n > 0 && n <= 100) break; // Valid input,exit loop
+   }
+   for (int i = 0; i < n; i++) {
+      printf ("Enter element %d: ", i + 1);
+      fgets (buffer, sizeof (buffer), stdin);
+      arr[i] = strtol (buffer, &endptr, 10);
+      if (endptr == buffer || *endptr != '\n') i--;
+   }
+   printf ("Original array: ");
+   for (int i = 0; i < n; i++) printf ("%d ", arr[i]);
+   printf ("\n");
+   InsertionSort (arr, n);
+   printf ("Sorted array: ");
+   for (int i = 0; i < n; i++) printf ("%d ", arr[i]);
+   printf ("\n");
+   while (1) {
+      char searchChoice;
+      printf ("Do you want Search (y|n)? : ");
+      searchChoice = _getch ();
+      printf ("%c\n", searchChoice);
+      if (searchChoice == 'y' || searchChoice == 'Y') {
+         printf ("Enter element to search: ");
+         fgets (buffer, sizeof (buffer), stdin);
+         int key = strtol (buffer, &endptr, 10), res = BinarySearch (arr, n, key);
+         if (endptr != buffer && *endptr == '\n') {
+            printf ((res != -1) ? "Element %d found at index %d\n"
+                    : "Element %d is not found\n", key, res);
+            break;
+         }
+         else printf (RED_TEXT "Invalid Input,enter valid number\n" RESET_TEXT);
+      }
+      else if (searchChoice == 'n' || searchChoice == 'N') break;
+      else printf (RED_TEXT "Invalid choice, please enter valid choice \n" RESET_TEXT);
+   }
+}
+
 int main () {
-   PrintHelloWorld ();
+   TestCases ();
+   char choice;  // Variable to hold a single character choice
+   while (1) {
+      printf ("\n1. User Input\n2. Exit the program\n"
+              "Enter your choice (1 or 2 ) : ");
+      choice = _getch ();
+      printf ("%c\n", choice);
+      system ("cls");
+      if (choice == '1' || choice == '2') {

```

Comment:
switch case has a special 'case' called 'default' that performs the same function.